Add UserGuide example of custom compiler flags  march=native build (#177)
MDAnalysis/UserGuide: doc/source/installation.rst

    pip install pytest-xdist
    pytest --disable-pytest-warnings --pyargs MDAnalysisTests --numprocesses 4

Optimised installations for specific CPU architectures
------------------------------------------------------

An experimental option for compiling the C/C++ in MDAnalysis with architecture specific optimisations enabled has been developed.

This can be activated by uncommenting the "march=native" option in `setup.cfg` and will result in the compiler using CPU architecture specific
optimisations on x86_64 (equivalent of `-march=native` compiler flag) and ARM platforms (equivalent of `-mcpu=native` compiler flag).

Note that currently, no additional optimisations are applied on Windows or if using an MSVC compiler.
Use of this option will result in performance gains where data and pipelining dependencies can be easily identified by the compiler and autovectorisation applied.

.. warning::
    Use of this option is considered **advanced** and will reduce the binary compatibility of MDAnalysis significantly,
    making it usable only on a matching CPU architecture to the one it is compiled on.

> 

Reviewer: Add in recommendation to run tests with the new build?

    pip install pytest-xdist
    pytest --disable-pytest-warnings --pyargs MDAnalysisTests --numprocesses 4

Custom compiler flags and optimised installations
-------------------------------------------------

You can pass any additional compiler flags for the C/C++ compiler using the ``extra_cflags`` variable in ``setup.cfg``.
This allows you to add any additional compiler options required for your architecture. 

> For example, ``extra_cflags`` can also be used to tune your MDAnalysis installation for your current architecture using the `-march`, `-mtune`, `-mcpu` and related compiler flags.

Reviewer 2: ```suggestion
For example, ``extra_cflags`` can be used to tune your MDAnalysis installation for your current architecture using the `-march`, `-mtune`, `-mcpu` and related compiler flags.
```

    pip install pytest-xdist
    pytest --disable-pytest-warnings --pyargs MDAnalysisTests --numprocesses 4

Custom compiler flags and optimised installations
-------------------------------------------------

> You can pass any additional compiler flags for the C/C++ compiler using the `extra_cflags` variable in `setup.cfg`.

Reviewer 2: ```suggestion
You can pass any additional compiler flags for the C/C++ compiler using the ``extra_cflags`` variable in ``setup.cfg``.
```

This allows you to add any additional compiler options required for your architecture. 

> This option can also be used to tune your MDAnalysis installation for your current architecture using `-march`, `-mtune`, `-mcpu` and related compiler flags.

Reviewer 2: ```suggestion
For example, ``extra_cflags`` can also be used to tune your MDAnalysis installation for your current architecture using the `-march`, `-mtune`, `-mcpu` and related compiler flags.
```

Which option?

> The use of these compiler flags allows CPU architecture specific optimisations. An example for an x86_64 machine would be to change the

Reviewer 2: ```suggestion
*Which* particular compiler flags to use depends on your CPU architecture. An example for an x86_64 machine would be to change the
```

> .. code-block:: YAML

Reviewer 2: I don't think this is yaml? Does a diff work in sphinx?

```
.. code-block:: diff

	- #extra_cflags = 
	+ extra_cflags = -march=native -mtune=native
```

    #extra_cflags =

line in `setup.cfg` to instead be:

.. code-block:: YAML

    extra_cflags = -march=native -mtune=native

Use of these flags can give a significant performance boost where the compiler can effectively autovectorise.

> Be sure to use the recommended flags for your target architecture. For example, ARM platforms reccomend against the use of `-mtune` in favour of `-mcpu`, while

Reviewer 2: ```suggestion
Be sure to use the recommended flags for your target architecture. For example, ARM platforms recommend using ``-mcpu`` *instead* of ``-mcpu``, while
```

> PowerPC platforms prefer both `-mcpu` and `-mtune`.

Reviewer 2: ```suggestion
PowerPC platforms prefer *both* ``-mcpu`` and ``-mtune``.
```

> Full dicussion of the these flags is available elsewhere and a list of supported options should be provided by your compiler. The list for GCC_ is provided here. 

Reviewer 2: > elsewhere

A link would be handy, or terms for googling.

.. warning::
    Use of these compiler options is considered **advanced** and may reduce the binary compatibility of MDAnalysis significantly, especially if using `-march`,

>     making it usable only on a matching CPU architecture to the one it is compiled on. We **strongly** recommend that you run the testsuite on your intended platform

Reviewer 2: ```suggestion
    making it usable only on a matching CPU architecture to the one it is compiled on. We **strongly** recommend that you run the test suite on your intended platform
```

>     before proceeding.

Reviewer 2: ```suggestion
    before proceeding with analysis.
```

Otherwise it sounds like we're to run the test suite before proceeding with compilation.

> For example, if you plan on using MDAnalysis on a heterogenous system, such as a supercomputer, where the login node you compile on and the
> compute node you run MDAnalysis on have different CPU architectures, you should avoid this option unless you know what you are doing, or you should compile directly on the compute node.

Reviewer 2: ```suggestion
In cases where you might encounter multiple CPU architectures (e.g. on a supercomputer where the login node and compute node have different architectures), you should avoid changing these options unless you are experienced with compiling software in these situations.
```

This is really long. It also implies that it's common to compile on the login node, whereas I feel like hpc people are really really *against* that kind of shenanigan.
